src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -5142,6 +5142,64 @@ public IList<Tensor> nonzero_as_list()
                 return t.chunk(t.shape[1], dim: 1);
             }
 
+            [DllImport("LibTorchSharp")]
+            extern static IntPtr THSTensor_roll(IntPtr tensor, IntPtr shifts, int shLength, IntPtr dims, long dimLength);
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll(long shifts, long? dims = null)
+            {
+                if (dims.HasValue) {
+                    return roll(new long[] { shifts }, new long[] { dims.Value });
+                }
+                else {
+                    return roll(new long[] { shifts }, null);
+                }
+            }
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll((long,long) shifts, (long,long) dims)
```

Comment:
>, [](http://example.com/codeflow?start=36&length=1)

nit: insert space.

Reply:
I wasn't able to make the 'dims' argument a span, since it didn't let me default to 'null' Is there a solution to that? An empty span?

Reply:
You can just pass empty span when having null dims. You can pass `default` value for the dims parameter.